### PR TITLE
add tests for relation validation lambda generation for dq

### PR DESCRIPTION
**Pull request by keenkeystrokes**
#### What type of PR is this?
- Improvement

#### What does this PR do / why is it needed ?

Adds tests for data quality lambda generation for relation validation

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

File: legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test_utils.pure [hunk 1]
```diff
@@ -63,7 +85,15 @@ function meta::external::dataquality::tests::assertLambdaEquals(expected:Functio
                                   ]
                             )
             );
+  assertEquals($expected->printFunctionDefinition($config, ''), $actual->printFunctionDefinition($config, ''));
```
Comment (gs-jp1) [inline legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test_utils.pure:88]: can we reuse the existing assertLambdaAndJSONEquals/assertLambdaEquals function?

File: legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test.pure [hunk 2]
```diff
@@ -170,4 +170,38 @@ function <<test.Test>> meta::external::dataquality::tests::testMetadata():Boolea
     
     //TODO assert on structure.
     assert($metadata->isNotEmpty());
+}
+
+function <<test.Test>> meta::external::dataquality::tests::testLambdaGeneration_relationValidation():Boolean[1]
+{
+  doRelationTest('###DataQualityValidation' +
```
Comment (gs-jp1) [inline legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test.pure:177]: can we simplify to reduce the boilerplate being passes in similar to the non-relation tests?

Comment (keenkeystrokes) [reply legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test.pure:177]: The code under test is the only input variable here which makes more sense to me than constructing the code under test using individual params

**Review by aormerod-gs — APPROVED** [review]

File: legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test_utils.pure (hunk 1 above)
Comment (keenkeystrokes) [reply legend-engine-xts-dataquality/legend-engine-xt-dataquality-pure-test/src/main/resources/core_dataquality_test/dataquality_test_utils.pure:86]: Used existing lambda equality function.

**Pull request merged**